src/Data/Collector.ts

```diff
@@ -63,6 +65,68 @@ export interface ReceiptVerificationResult {
   nestedCounterMessages?: string[]
 }
 
+/**
+ * Fetches authorized signers that belong to the execution shard from the provided signatures.
+ * @param {Crypto.core.Signature[] | P2PTypes.P2PTypes.Signature[]} signs - The array of signatures to verify.
+ * @param {CycleShardData} cycleShardData - The cycle shard data containing node information.
+ * @param {number} homePartition - The home partition number.
+ * @param {string} txId - The transaction ID.
+ * @param {number} timestamp - The timestamp of the transaction.
+ * @param {number} cycle - The cycle number.
+ * @returns {Set<[number, Crypto.core.Signature]> | Set<[number, P2PTypes.P2PTypes.Signature]>} - A set of tuples containing the index and signature of authorized signers.
+ */
+function fetchAuthorizedSigners(
+  signaturePack: Crypto.core.Signature[] | P2PTypes.P2PTypes.Signature[],
+  cycleShardData: CycleShardData,
+  homePartition: number,
+  txId: string,
+  timestamp: number,
+  cycle: number
+): Set<[number, Crypto.core.Signature]> | Set<[number, P2PTypes.P2PTypes.Signature]> {
+
+  const nodeMap = new Map<string, P2PTypes.NodeListTypes.Node>()
+  // Fill the map with nodes keyed by their public keys
+  cycleShardData.nodes.forEach((node) => {
+    if (node.publicKey) {
+      nodeMap.set(node.publicKey, node)
+    }
+  })
+
+  // Create a set to store acceptable signers
+  // in case of GlobalTxReceipt, the signatures are of type : P2PTypes.P2PTypes.Signature
+  // in case of NonGlobalTxReceipt, signatures are of type : Crypto.core.Signature
+  const acceptableSigners = new Set<[number, Crypto.core.Signature | P2PTypes.P2PTypes.Signature]>()
+
+  for (const [index, sign] of signaturePack.entries()) {
+    const { owner: nodePubKey } = sign
+    // Get the node id from the public key
+    const node = nodeMap.get(nodePubKey)
+
+    // If the node is not found in the active nodes list, log an error and continue
+    if (node == null) {
+      Logger.mainLogger.error(
+        `The node with public key ${nodePubKey} of the receipt ${txId} with ${timestamp} is not in the active nodesList of cycle ${cycle}`
+      )
+      if (nestedCountersInstance)
+        nestedCountersInstance.countEvent('receipt', 'sign_owner_not_in_active_nodesList')
+      continue
+    }
+
+    // Check if the node is in the execution group
+    if (!cycleShardData.parititionShardDataMap.get(homePartition).coveredBy[node.id]) {
+      Logger.mainLogger.error(
+        `The node with public key ${nodePubKey} of the receipt ${txId} with ${timestamp} is not in the execution group of the tx`
+      )
+      if (nestedCountersInstance)
+        nestedCountersInstance.countEvent('receipt', 'node_not_in_execution_group_of_tx')
+      continue
+    }
+
+    acceptableSigners.add([index, sign])
```

Comment:
is `node.publicKey` as good as unique signers? if so, we should lowercase and calculate uniqueness. There have been some reports that same value but different case (e.g. 'abcd', 'Abcd', 'aBcd') are considered as different but they are actually the same.